tests/test_unittest/test_gtscript_frontend.py

```diff
@@ -241,6 +229,37 @@ def definition_func(inout_field: gtscript.Field[float]):
             )
 
 
+class TestAxisSyntax:
+    def test_good_syntax(self):
+        def definition_func(in_field: gtscript.Field[float], out_field: gtscript.Field[float]):
+            with computation(PARALLEL), interval(...):
+                out_field = in_field[I + 1, J - 1]
+
+        parse_definition(
+            definition_func, name=inspect.stack()[0][3], module=self.__class__.__name__
+        )
+
+    def test_bad_mul_syntax(self):
```

Comment:
I would also test using indices in fields without that axis and using repeated axis names like `field[I+1, I+3]`